docs/src/pages/components/data-grid/filtering/QuickFilteringGrid.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import { IconButton, TextField } from '@material-ui/core';
+import {
+  DataGrid,
+  GridToolbarColumnsButton,
+  GridToolbarContainer,
+  GridToolbarDensitySelector,
+  GridToolbarFilterButton,
+} from '@material-ui/data-grid';
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+import ClearIcon from '@material-ui/icons/Clear';
+import SearchIcon from '@material-ui/icons/Search';
+
+const QuickSearchToolbar = (props) => {
+  return (
+    <GridToolbarContainer
+      style={{
+        justifyContent: 'space-between',
+      }}
+    >
+      <div>
+        <GridToolbarColumnsButton />
+        <GridToolbarFilterButton />
+        <GridToolbarDensitySelector />
+      </div>
+      <TextField
+        value={props.value}
+        onChange={props.onChange}
+        placeholder="Search..."
+        InputProps={{
+          startAdornment: <SearchIcon fontSize="small" />,
+          endAdornment: (
+            <IconButton
+              size="small"
+              style={{ visibility: props.value ? 'visible' : 'hidden' }}
+              onClick={props.clearSearch}
+            >
+              <ClearIcon fontSize="small" />
+            </IconButton>
+          ),
+        }}
+      />
+    </GridToolbarContainer>
+  );
+};
+
+export default function QuickFilteringGrid() {
+  const { data } = useDemoData({
+    dataSet: 'Commodity',
+    rowLength: 100,
+    maxColumns: 6,
+  });
+  const [searchText, setSearchText] = React.useState('');
+  const [rows, setRows] = React.useState<any[]>(data.rows);
+
+  const requestSearch = (searchValue: string) => {
+    setSearchText(searchValue);
+    const searchRegex = new RegExp(`.*${searchValue}.*`, 'ig');
```

Comment:
regexp value to escape?

Reply:
Should we ignore diatrics too in the match?

Reply:
For context, in the main repo, we have used two different search libraries so far: 

- [match-sorter](https://github.com/kentcdodds/match-sorter) in https://next.material-ui.com/components/autocomplete/#advanced
- [flexsearch](https://github.com/nextapps-de/flexsearch) in https://next.material-ui.com/components/material-icons/

I wonder how far we should go. If we don't go very far, how about we encourage a library to go beyond the demo, so our users don't have to think?

Reply:
We don't need to ignore diacritics because the strings are all ASCII. I would keep it very simple, using even `row[k].toString().indexOf(...)`. If we keep the Regex approach, then at least let's escape special characters. We could add a note pointing to these search libraries.

```suggestion
    const searchRegex = new RegExp(escapeRegExp(searchValue), 'i');
```